### PR TITLE
feat: support usedb hint for customize the sql routing

### PR DESCRIPTION
**Pull request by dk-lockdown**
ref: https://github.com/cectc/dbpack/issues/13

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
there are unit test cases

### Ⅳ. Describe how to verify it
check ci result

### CODE REVIEW

File: pkg/executor/read_write_splitting.go [hunk 1]
```diff
@@ -29,7 +30,13 @@ import (
 	"github.com/cectc/dbpack/pkg/log"
 	"github.com/cectc/dbpack/pkg/mysql"
 	"github.com/cectc/dbpack/pkg/proto"
+	"github.com/cectc/dbpack/pkg/resource"
 	"github.com/cectc/dbpack/third_party/parser/ast"
+	"github.com/cectc/dbpack/third_party/parser/model"
+)
+
+const (
+	hintUseDB = "UseDB"
```
Comment (bohehe) [inline pkg/executor/read_write_splitting.go:40]: Is there any hintType we can use?

**Comment by adofsauron** [comment]
LGTM

File: pkg/executor/read_write_splitting.go [hunk 2]
```diff
@@ -221,14 +228,35 @@ func (executor *ReadWriteSplittingExecutor) ExecutorComQuery(ctx context.Context
...
+		txi, ok := executor.localTransactionMap.Load(connectionID)
+		if ok {
+			// in local transaction
+			tx = txi.(proto.Tx)
+			return tx.Query(ctx, sql)
+		}
+		if has, dsName := hasUseDBHint(stmt.TableHints); has {
+			protoDB := resource.GetDBManager().GetDB(dsName)
+			if protoDB == nil {
+				log.Debugf("data source %d not found", dsName)
```
Comment (bohehe) [inline pkg/executor/read_write_splitting.go:252]:
```suggestion
				log.Debugf("data source %s not found", dsName)
```

File: pkg/executor/read_write_splitting.go [hunk 3]
```diff
@@ -241,15 +269,27 @@ func (executor *ReadWriteSplittingExecutor) ExecutorComStmtExecute(ctx context.C
...
 	case *ast.InsertStmt, *ast.DeleteStmt, *ast.UpdateStmt:
 		db := executor.masters.Next(proto.WithMaster(ctx)).(*DataSourceBrief)
 		return db.DB.ExecuteStmt(proto.WithMaster(ctx), stmt)
 	case *ast.SelectStmt:
-		db := executor.reads.Next(proto.WithSlave(ctx)).(*DataSourceBrief)
+		var db *DataSourceBrief
+		if has, dsName := hasUseDBHint(st.TableHints); has {
+			protoDB := resource.GetDBManager().GetDB(dsName)
+			if protoDB == nil {
+				log.Debugf("data source %d not found", dsName)
```
Comment (bohehe) [inline pkg/executor/read_write_splitting.go:291]:
```suggestion
				log.Debugf("data source %s not found", dsName)
```

**Review by bohehe — CHANGES_REQUESTED** [review]

File: pkg/executor/read_write_splitting.go (hunk 1 above)
Comment (dk-lockdown) [reply pkg/executor/read_write_splitting.go:40]: hint defined by https://github.com/CECTC/dbpack/blob/dev/third_party/parser/hintparser.y

**Review by bohehe — APPROVED** [review]
LGTM

**Pull request merged**